guides/v2.2/contributor-guide/contributing_dod.md

```diff
@@ -57,74 +57,115 @@ Any backward-incompatible changes must also be recorded in the accompanying docu
+
+Assuming there is a method with the signature `function removeLetterFromString(string $letter, string $string): string`
+
+- A blackbox test would test the obvious cases both normal cases such as `removeLetterFromString('a', 'fooabar') === 'foobr'` and edge cases such as when `$letter` or `$string` are empty or `$letter` contains multiple or multibyte characters. It would probably test several variations of string lengths. Notice there is no knowledge of how this function is written.
+- A whitebox test would look into the implementation of the method and see that it calls a microservice API and make sure that each condition of the code has test coverage. For example, this test may force the microservice client to return an error and have a test for the expected handling of that scenario. This wouldn't be a reusable test and it's directly coupled to the implementation. 
+- A graybox test would look at the implementation of the method and see that it calls a microservice API and injects the `$string` into the URL so it may add some extra test coverage for strings that contain unsafe URL characters such as `&?=%/`. This isn't coupled to the implementation as the implementation could be changed and the test should still pass.   
+
 ### Automated Tests
 
 Code changes must be covered by automated tests according to Classification of Magento Automated Tests.
 When choosing how to cover your changes, pick the most lightweight (execution time wise) test type that will provide sufficient coverage.
 If you encounter an existing test that insufficiently covers your changes, you can delete that test but you must write a proper test to replace it. For example, a method that interacts with database has a unit test. You can replace it with an integration test.
 
-Before committing code changes, author must ensure successful execution of all tests by running all tests or at least those which might be affected by code changes.
-Continuous integration enforces execution of all tests and author is accountable for broken builds.
+Be aware that while high-level tests may provide coverage to code, it is only indirect coverage. Tests with more direct usage of the changed code will likely still needs to be written to ensure that regardless of the which components are actually used in the black box, the individual components still have coverage.  
+
+High level testing policy:
+ 
+* Before committing code changes, author must ensure successful execution of all tests by running all tests or at least those which might be affected by code changes.
+  Continuous integration enforces execution of all tests and author is accountable for broken builds.
+* Any code that is added or changed must have explicit tests to validate the behavior. To clarify, this policy is intended to prove that the integrity of the code is verified regardless of which implementations are loaded in a black-box environment. This may be done through unit or integration testing as appropriate. 
```

Comment:
May be confusing to hyphenate blackbox

Reply:
I meant to go back and fix this. Thank you!